### PR TITLE
Combining all changes that fix flakiness

### PR DESCRIPTION
**Pull request by andresuribe87**
This change combines https://github.com/TBD54566975/ssi-service/pull/321 and https://github.com/TBD54566975/ssi-service/pull/318 because they were both causes for flakiness in e2e tests. Refer to those PRs if you want to see isolated changes.

### CODE REVIEW

File: pkg/testutil/well_known_generation_test.go [hunk 1]
```diff
@@ -72,7 +72,7 @@ func createWellKnownDIDConfiguration(tt *testing.T, didResponse *did.CreateDIDRe
 	err = builder.SetCredentialSubject(subject)
 	assert.NoError(tt, err)
 
-	err = builder.AddContext("https://identity.foundation/.well-known/did-configuration/v1")
+	err = builder.AddContext("https://tbd.website/.well-known/did-configuration/v1")
```
Comment (nitro-neal) [inline pkg/testutil/well_known_generation_test.go:75]: love it

File: pkg/testutil/well_known_generation_test.go [hunk 2]
```diff
@@ -137,7 +137,7 @@ func createWellKnownDIDConfiguration(tt *testing.T, didResponse *did.CreateDIDRe
 	credToken := keyaccess.JWT(signedTokenBytes).Ptr()
 
 	wellKnown := map[string]interface{}{
-		"@context":    "https://identity.foundation/.well-known/did-configuration/v1",
+		"@context":    "https://tbd.website/.well-known/did-configuration/v1",
```
Comment (nitro-neal) [inline pkg/testutil/well_known_generation_test.go:140]: this is 404 btw

**Review by nitro-neal — APPROVED** [review]
Awesome thanks for doing this!

File: pkg/service/credential/storage.go [hunk 3]
```diff
@@ -623,29 +616,24 @@ func (cs *Storage) GetStatusListCurrentIndexWatchKey(issuer, schema, statusPurpo
...
 	storedStatusListCreds, err := cs.GetStatusListCredentialsByIssuerSchemaPurpose(ctx, issuer, schema, statusPurpose)
 	if err != nil {
-		return "", nil, util.LoggingNewErrorf("getting status list credential for issuer: %s schema: %s", issuer, schema)
+		return nil, util.LoggingNewErrorf("getting status list credential for issuer: %s schema: %s", issuer, schema)
 	}
 
 	// This should never happen, there should always be only 1 status list credential per <issuer,schema, statusPurpose> triplet
 	if len(storedStatusListCreds) > 1 {
-		return "", nil, util.LoggingNewErrorf("only one status list credential per <issuer,schema> pair allowed. issuer: %s schema: %s", issuer, schema)
+		return nil, util.LoggingNewErrorf("only one status list credential per <issuer,schema> pair allowed. issuer: %s schema: %s", issuer, schema)
```
[the remark this answers is not in the document]
Comment (andresuribe87) [reply pkg/service/credential/storage.go:627]: I'm not sure if the error you were seeing was a root cause, or a side effect.

We [should not panic](https://github.com/uber-go/guide/blob/master/style.md#dont-panic). In this case, if this situation is truly a bad state, we should emit a FATAL error. I don't see this warranting stopping the execution of ssi-service. Happy to talk through this if you think we should!

File: pkg/testutil/well_known_generation_test.go (hunk 2 above)
Comment (andresuribe87) [reply pkg/testutil/well_known_generation_test.go:140]: Good point. I'm going to leave as is. This file seems to be ignored in tests, and appears to be meant as an example only.

File: integration/credential_revocation_concurrency_integration_test.go [hunk 4]
```diff
@@ -111,17 +111,25 @@ func TestConcurrencyRevocationVerifiableCredentialIntegration(t *testing.T) {
 	assert.NotEmpty(t, schemaID)
 
 	const vcCount = 100
-	credStatusListIndexes := make([]string, vcCount)
+	credStatusListIndexes := make([]string, 0, vcCount)
 
 	var wg sync.WaitGroup
 	wg.Add(vcCount)
 
+	m := sync.Mutex{}
```
Comment (decentralgabe) [inline integration/credential_revocation_concurrency_integration_test.go:119]: var m sync.Mutex

File: integration/credential_revocation_concurrency_integration_test.go [hunk 5]
```diff
@@ -111,17 +111,25 @@ func TestConcurrencyRevocationVerifiableCredentialIntegration(t *testing.T) {
 	assert.NotEmpty(t, schemaID)
 
 	const vcCount = 100
-	credStatusListIndexes := make([]string, vcCount)
+	credStatusListIndexes := make([]string, 0, vcCount)
```
Comment (decentralgabe) [inline integration/credential_revocation_concurrency_integration_test.go:114]: if 0 init does this syntax give us a benefit over `var credStatusListIndexes []string`?

File: pkg/service/credential/storage.go [hunk 6]
```diff
@@ -623,29 +616,24 @@ func (cs *Storage) GetStatusListCurrentIndexWatchKey(issuer, schema, statusPurpo
...
 	}
 
 	// No Status List Credential Exists, create a new uuid
 	if storedStatusListCreds == nil || len(storedStatusListCreds) == 0 {
 		logrus.Warn("No Status List Credential Exists, create a new uuid")
```
Comment (decentralgabe) [inline pkg/service/credential/storage.go:632]: would be useful to log the schema and issuer here

File: pkg/testutil/well_known_generation_test.go (hunk 1 above)
Comment (decentralgabe) [inline pkg/testutil/well_known_generation_test.go:75]: test util is for integration tests only? if not we should rethink this

File: integration/credential_revocation_concurrency_integration_test.go (hunk 5 above)
Comment (andresuribe87) [reply integration/credential_revocation_concurrency_integration_test.go:114]: Yes, the correct capacity is allocated. More details [here](https://github.com/uber-go/guide/blob/master/style.md#specifying-slice-capacity)

File: pkg/testutil/well_known_generation_test.go (hunk 1 above)
Comment (andresuribe87) [reply pkg/testutil/well_known_generation_test.go:75]: testutil is currently being skipped right now. I'm of the opinion that we should remove, and add some of this functionality to a CLI.

File: integration/credential_revocation_concurrency_integration_test.go (hunk 4 above)
Comment (andresuribe87) [reply integration/credential_revocation_concurrency_integration_test.go:119]: Adding in another PR.

**Pull request merged**